compiler/analyzer.ml

```diff
@@ -424,7 +490,45 @@ and check_binop env e1 op e2 =
         let env', ew2' = constrain_ew env' ew2 Bool in
         env', Sast.Expr(Sast.Binop(ew1, op, ew2), Bool)
       else binop_error typ1 op typ2
-
+    | Dplus | Dtimes as op ->
+      if is_dist typ1 && is_dist typ2 then 
+        let result_type = Dist_t in  
```

Comment:
nit: we probably don't need this variable if we're not permuting it. we can just write `Dist_t` directly in the `Sast.Expr`s